### PR TITLE
minor lint oriented cleanup and refactor in headers

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/packet_pool.hpp [hunk 1]
```diff
@@ -168,7 +168,7 @@ namespace libtorrent {
 			, m_mtu_floor_slab(mtu_floor_size)
 			, m_mtu_ceiling_slab(mtu_ceiling_size)
 		{}
-		packet_pool(packet_pool&&) = default;
+		packet_pool(packet_pool&&) = delete;
```
Comment (arvidn) [inline include/libtorrent/packet_pool.hpp]: I don't understand this change

Comment (arvidn) [reply include/libtorrent/packet_pool.hpp]: is there a non-movable member in here?

Comment (arvidn) [reply include/libtorrent/packet_pool.hpp]: I see, I would prefer making ``packet_slab`` movable instead, by just defaulting its move constructor.

Comment (aldenml) [reply include/libtorrent/packet_pool.hpp]: yes, the packet_slab members, they have `packet_slab(const packet_slab&) = delete;`

Comment (aldenml) [reply include/libtorrent/packet_pool.hpp]: ok, I will change that

**Pull request merged**